azkaban-common/src/main/java/azkaban/executor/ExecutionRampDao.java

```diff
@@ -525,11 +576,7 @@ public void updateExecutedRampFlows(final String ramp, ExecutableRampExceptional
 
     try {
 
-      Object[][] parameters = executableRampExceptionalItems
-          .getItems()
-          .entrySet()
-          .stream()
-          .filter(item -> item.getValue().isCachedOnly())
+      Object[][] parameters = executableRampExceptionalItems.getCachedItems().stream()
           .map(item -> {
             ArrayList<Object> object = new ArrayList<>();
```

Comment:
Should it be List instead of ArrayList on LHS?

Reply:
No. It need to convert to array by toArray(). It will be better to keep ArrayList instead.